src/cursor/cur_stat.c

```diff
@@ -460,9 +460,8 @@ __curstat_layered_init(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR_STAT
     WT_ERR(__init_layered_constituent_stats(session, cst));
     WT_ERR(__wt_session_release_dhandle(session));
 
-    __wt_curstat_dsrc_final(cst);
-
 done:
+    __wt_curstat_dsrc_final(cst);
```

Comment:
This had to be moved after the done label as it was causing segfaults.
In follower mode, if no stable table exists, we jump to done, meaning we skip gathering stats from the stable table. However, we still care about ingest tables and want to ensure the stat structure is properly finalised for them.